mlir/lib/ExecutionEngine/ROCm/BackendUtils.cpp

```diff
@@ -41,17 +41,94 @@
 
 #include <mutex>
 #include <numeric>
+#include <string.h>
 
 using namespace mlir;
 
 static constexpr const char kRunnerProgram[] = "mlir-rocm-runner";
 static constexpr const char kRocmAgentEnumerator[] = "rocm_agent_enumerator";
+static constexpr const char gcnArchDelimiter[] = ":";
+static constexpr const char targetTriple[] = "amdgcn-amd-amdhsa";
+
+namespace {
+// TODO: remove this once the rocm_agent_enumerator is ready
+void getGpuGCNArchName(hipDevice_t device, std::string &gcnArchName) {
+  hipDeviceProp_t props;
+  hipError_t result = hipGetDeviceProperties(&props, device);
```

Comment:
Although I don’t feel like introducing HIP runtime dependency to a compiler, we’ll live with it for now

Reply:
Client wouldn't be using the hip runtime to get such information. This is only used by `mlir-rocm-runner`. As a library, it will only rely on the client passed in archname.

Reply:
Then wouldn't it make more sense to have `mlir-rocm-runner` do these queries?

Reply:
@krzysz00 The problem is that hip related functions are conceptually the "runtime" (This component is the only dependency that depend on hip runtime, refer to rocm-runtime-wrappers.cpp), whereas `mlir-rocm-runner` is a super thin application wrapper, with which we want to put in as minimal implementation as possible.

Reply:
Ok, so, is this called in situations analogous to where you have `-mtune=native` on a clang invocation? That is, we're needing to discover the current GPU architecture because someone's building for the machine they're running on right now?

Reply:
I think so. The only use case we had for `mlir-rocm-runner` is to run on the GPU:0 of local machine. I'm not aware of any unit test that try to use it a different way.

Reply:
In which case, "if the client hasn't called for an architecture, discover what they have" is understandable.